### PR TITLE
[2.0.x] Implement SDIO support for STM32F1

### PR DESCRIPTION
**Pull request by jmz52**
### Description

Add support for SDIO interface for STM32F1.

### Benefits

Marlin can used onboard SD cards on MKS Robin and various STM32F1 development boards.

### Related Issues

https://github.com/MarlinFirmware/Marlin/issues/11641

### Additional Information
Two defines are needed to enable SDIO interface
`#define SDSUPPORT`
`#define SDIO_SUPPORT`

To add SDIO support for other HALs three functions should be implemented in these HALs
`bool SDIO_Init(void);`
`bool SDIO_ReadBlock(uint32_t block, uint8_t *dst);`
`bool SDIO_WriteBlock(uint32_t block, const uint8_t *src);`

### CODE REVIEW

File: Marlin/src/HAL/HAL_STM32F1/HAL_sdio_STM32F1.cpp [hunk 1]
```diff
@@ -0,0 +1,279 @@
...
+
+  sdio_begin();
+  sdio_set_dbus_width(SDIO_CLKCR_WIDBUS_1BIT);
+
+  dma_init(SDIO_DMA_DEV);
+  dma_disable(SDIO_DMA_DEV, SDIO_DMA_CHANNEL);
+  dma_set_priority(SDIO_DMA_DEV, SDIO_DMA_CHANNEL, DMA_PRIORITY_VERY_HIGH);
+
+
+  if (!SDIO_CmdGoIdleState()) { return false; }
```
Comment (xC0000005) [inline Marlin/src/HAL/HAL_STM32F1/HAL_sdio_STM32F1.cpp]: If the intention is to ignore the result of the first, doesn't the return false here defeat that? (sorry if I misunderstand what is supposed to happen here).

Comment (jmz52) [reply Marlin/src/HAL/HAL_STM32F1/HAL_sdio_STM32F1.cpp]: CMD0 does not wait for SD card's response. 
False here means that SDIO hardware failed to send command, which should never be the case when SDIO is properly configured and clocked.

**Comment by xC0000005** [comment]
This method is working for me to read data using SDIO on lerdge boards as well (in the STM32 HAL), so very useful.

**Comment by Phr3d13** [comment]
OMG! Thank you so much for this PR @jmz52! This PR made my GTM32Pro board able to read the reprap discount smart display's SD card. There's only one more hurdle for my board (eeprom). Thank you, thank you, thank you, thank you!

**Pull request merged**